src/sources/file.rs

```diff
@@ -250,6 +250,13 @@ pub struct FileConfig {
     #[configurable(derived)]
     #[serde(default)]
     internal_metrics: FileInternalMetricsConfig,
+
+    /// How long to keep an open handle to a rotated log file.
+    /// The default value represents "no limit"
+    #[serde_as(as = "serde_with::DurationSeconds<u64>")]
+    #[configurable(metadata(docs::type_unit = "seconds"))]
+    #[serde(default = "default_rotate_wait_sec")]
+    pub rotate_wait_sec: Duration,
```

Comment:
Small nit - Similar to [other places](https://github.com/vectordotdev/vector/blob/e8a09d996f363fa12b87e38f8e493101b9e2a605/src/sinks/file/mod.rs#L65) in the codebase, we will want to name this field `rotate_wait` because its type is a Duration (`_secs` is not relevant in the subsequent code) and we will use `serde(rename)` to specify the `_secs` prefix for the user facing config.

```suggestion
    /// How long to keep an open handle to a rotated log file.
    /// The default value represents "no limit".
    #[serde_as(as = "serde_with::DurationSeconds<u64>")]
    #[configurable(metadata(docs::type_unit = "seconds"))]
    #[serde(default = "default_rotate_wait", rename = "rotate_wait_secs")]
    pub rotate_wait: Duration,
```